### PR TITLE
Powershell : Release for MicrosoftCache.RedisEnterprise - 2024-09-01 PREVIEW

### PR DESCRIPTION
**Pull request by priyankarking**
…Preview

## Description

Change log:
- Adds support for using Microsoft Entra token-based authentication.
- Cluster has new properties: highAvailability and redundancyMode.
- New product SKUs added.
- Database has new properties:  accessKeysAuthentication. 

Powershell : Release for MicrosoftCache.RedisEnterprise - 2024-09-01 PREVIEW
Add support for access via MS Entra ID feature.
Added new product SKUs.
Added new properties w.r.t cluster

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+### -UserObjectId
+The object ID of the user.
+
+```yaml
+Type: System.String
+Parameter Sets: CreateExpanded, CreateViaIdentityExpanded
+Aliases:
+
+Required: False
```
Comment (priyankarking) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md:252]: How to make this a required field ?

**Comment by msJinLei** [comment]
@priyankarking  Please record the following test cases
```
Describing Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink
    [-] Force unlink database from group 39ms
      ArgumentException: Missing Request 'Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink+[NoContext]+Force unlink database from group+$POST+[https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test3-6jqyti5f/databases/default/forceUnlink?api-version=2024-09-01-preview+1'](https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test3-6jqyti5f/databases/default/forceUnlink?api-version=2024-09-01-preview+1%27) in recording file (Parameter 'rqKey')
      at Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\exports\ProxyCmdletDefinitions.ps1: line 1099
      at <ScriptBlock>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\test\Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink.Tests.ps1: line 26

Executing script D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\test\New-AzRedisEnterpriseCache.Tests.ps1
Running 'playback' mode...
Using recording D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\test\New-AzRedisEnterpriseCache.Recording.json

  Describing New-AzRedisEnterpriseCache
    [-] Create 84ms
      ArgumentException: Missing Request 'New-AzRedisEnterpriseCache+[NoContext]+Create+$PUT+[https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test1-b7q1ojzt?api-version=2024-09-01-preview+1'](https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test1-b7q1ojzt?api-version=2024-09-01-preview+1%27) in recording file (Parameter 'rqKey')
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\internal\ProxyCmdletDefinitions.ps1: line 1853
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\custom\New-AzRedisEnterpriseCache.ps1: line 324
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\exports\ProxyCmdletDefinitions.ps1: line 3592
      at <ScriptBlock>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\test\New-AzRedisEnterpriseCache.Tests.ps1: line 29
    [-] CreateNoDatabase 29ms
      ArgumentException: Missing Request 'New-AzRedisEnterpriseCache+[NoContext]+CreateNoDatabase+$PUT+[https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test2-a0lvx1bq?api-version=2024-09-01-preview+1'](https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test2-a0lvx1bq?api-version=2024-09-01-preview+1%27) in recording file (Parameter 'rqKey')
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\internal\ProxyCmdletDefinitions.ps1: line 1853
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\custom\New-AzRedisEnterpriseCache.ps1: line 324
      at New-AzRedisEnterpriseCache<Process>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\exports\ProxyCmdletDefinitions.ps1: line 3592
      at <ScriptBlock>, D:\a\_work\1\s\artifacts\Debug\Az.RedisEnterpriseCache\RedisEnterpriseCache.Autorest\test\New-AzRedisEnterpriseCache.Tests.ps1: line 59
ps-cache-test3-6jqyti5f
    [-] Create a cache without a database to create a georeplicated database later 33ms
      ArgumentException: Missing Request 'New-AzRedisEnterpriseCache+[NoContext]+Create a cache without a database to create a georeplicated database later+$PUT+[https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test3-6jqyti5f?api-version=2024-09-01-preview+1'](https://management.azure.com/subscriptions/b3e4e86c-59f5-46c5-8b86-e97530eac8fb/resourceGroups/ps-redisenterprise-rg-i1awo54c/providers/Microsoft.Cache/redisEnterprise/ps-cache-test3-6jqyti5f?api-version=2024-09-01-preview+1%27) in recording file (Parameter 'rqKey')
```

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+### CreateExpanded (Default)
+```
+New-AzRedisEnterpriseCacheAccessPolicyAssignment -AccessPolicyAssignmentName <String> -ClusterName <String>
+ -DatabaseName <String> -ResourceGroupName <String> [-SubscriptionId <String>] [-AccessPolicyName <String>]
+ [-UserObjectId <String>] [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-Confirm] [-WhatIf]
+ [<CommonParameters>]
+```
+
+### Create
```
Comment (msJinLei) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md:23]: Please remove the parameter sets `Create` and `CreateViaIdentity`

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md (hunk 1 above)
Comment (msJinLei) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/New-AzRedisEnterpriseCacheAccessPolicyAssignment.md:243]: You can customize the cmdlet https://github.com/Azure/autorest.powershell/blob/main/docs/customization.md

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please upload *.Recording.json files of your test cases

**Comment by msJinLei** [comment]
Please also provide a change log for your PR

**Comment by priyankarking** [comment]
> Please upload *.Recording.json files of your test cases

@msJinLei :actual tests and recording will be added at the time of GA release. For now I have tested it manually.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**